decidim-core/lib/decidim/icon_registry.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Decidim
+  class IconRegistry
+    def initialize
+      @icons = {}
+    end
+
+    def register(name:, icon:, resource:, description:, category:)
+      @icons[name] = OpenStruct.new(name:, icon:, resource:, description:, category:)
```

Comment:
Heads-up, since Ruby 3 we should not use OpenStruct. It isn't flagged by rubocop because [we have it disabled](https://github.com/decidim/decidim/blob/b1bab1b3227cd97a813b3078fbcd32f3bd855387/decidim-dev/config/rubocop/disabled.yml#L1), but we should at least try to not introduce new code using it.